// Authors: Chen Qian(qcdsr970209@gmail.com)

syntax = "proto2";

option cc_generic_services = true;

package ctgfs;

message ClientKVRequest {
    required uint64 id = 1;
    required string addr = 2;

    message Command {
        enum Type {
            kCreateDir = 0;
            kRemoveDir = 1;
            kWriteFile = 2;
            kRemoveFile = 3;
            kReadFile = 4;
        }
        required Type type = 2;
        required string path = 3;
        optional string value = 4;
    }
    optional Command command = 3;
};

message ClientKVResponse {
	required uint64 id = 1;
	required string addr = 2;
};

service MasterService{
	rpc ClientAskForKV(ClientKVRequest) returns (ClientKVResponse);	
};
